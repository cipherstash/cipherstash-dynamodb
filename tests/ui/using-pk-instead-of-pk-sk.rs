use cipherstash_dynamodb::{Decryptable, Encryptable, Searchable};
use cipherstash_dynamodb::EncryptedTable;

#[derive(Debug, Encryptable, Decryptable, Searchable)]
struct User {
    #[cryptonamo(query = "exact")]
    #[partition_key]
    email: String,

    #[cryptonamo(query = "exact")]
    #[sort_key]
    name: String,
}

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let config = aws_config::from_env()
        .endpoint_url("http://localhost:8000")
        .load()
        .await;

    let client = aws_sdk_dynamodb::Client::new(&config);
    let table = EncryptedTable::init(client, "users").await?;

    let user: Option<User> = table.get("user@example.com").await?;

    Ok(())
}
