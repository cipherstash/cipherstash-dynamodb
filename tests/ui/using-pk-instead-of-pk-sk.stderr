error[E0277]: the trait bound `PkSk: From<&str>` is not satisfied
  --> tests/./ui/using-pk-instead-of-pk-sk.rs:25:40
   |
25 |     let user: Option<User> = table.get("user@example.com").await?;
   |                                    --- ^^^^^^^^^^^^^^^^^^ the trait `From<&str>` is not implemented for `PkSk`
   |                                    |
   |                                    required by a bound introduced by this call
   |
   = help: the trait `From<(Pk, Sk)>` is implemented for `PkSk`
   = note: required for `&str` to implement `Into<PkSk>`
note: required by a bound in `EncryptedTable::get`
  --> src/encrypted_table/mod.rs
   |
   |     pub async fn get<T>(&self, k: impl Into<T::PrimaryKey>) -> Result<Option<T>, GetError>
   |                                        ^^^^^^^^^^^^^^^^^^^ required by this bound in `EncryptedTable::get`
